data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java

```diff
@@ -254,6 +254,38 @@ public String toJsonString() {
         return jsonNode.toString();
     }
 
+    /**
+     * returns a string with formatted parts replaced by their values. The input
+     * string may contain parts with format "${.../.../...}" which are replaced
+     * by their value in the event
+     * @param format string with format
+     * @throws RuntimeException if the format is incorrect or the value is not a string
+     */
+    @Override
+    public String formatString(final String format) {
+        int fromIndex = 0;
+        String result = "";
+        int pos = 0;
```

Comment:
Please prefer full works for variable names. It's not too big of a deal with variables within functions. But, since I have some comments here anyway it might be nice to clean these up.